update soname to 40

JacobBarthelmeh opened the pull request:
# Description

Updates the library soname to 40. In wolfSSL release 5.6.0 there was three parameter type changes in public facing API that require that the age be reset. This is with an expanded configure used when tracking across version (--enable-distro --disable-examples).

```
[−] wolfSSL_EC_KEY_set_conv_form ( WOLFSSL_EC_KEY* eckey, char form )  1 
⇣
wolfSSL_EC_KEY_set_conv_form ( WOLFSSL_EC_KEY* key, int form )
```

```
[−] wolfSSL_EC_POINT_point2bn ( WOLFSSL_EC_GROUP const* group, WOLFSSL_EC_POINT const* p, char form, WOLFSSL_BIGNUM* in, WOLFSSL_BN_CTX* ctx )  1 
⇣
wolfSSL_EC_POINT_point2bn ( WOLFSSL_EC_GROUP const* group, WOLFSSL_EC_POINT const* point, int form, WOLFSSL_BIGNUM* bn, WOLFSSL_BN_CTX* ctx )
```

```
[−] wolfSSL_EC_POINT_point2oct ( WOLFSSL_EC_GROUP const* group, WOLFSSL_EC_POINT const* p, char form, byte* buf, size_t len, WOLFSSL_BN_CTX* ctx )  1 
⇣
wolfSSL_EC_POINT_point2oct ( WOLFSSL_EC_GROUP const* group, WOLFSSL_EC_POINT const* point, int form, byte* buf, size_t len, WOLFSSL_BN_CTX* ctx )
```

## How did you test?

Using abi-tracker and abi-monitor

dgarske reviewed (commented):
We should add a comment that this needs to be updated due to the Debian packaging and put in the release procedure. I mentioned this at the end of the release, but did not realize the impact of not updating it...

JacobBarthelmeh commented:
Added an improved internal script for checking. Expanding it to now check --enable-distro where before on the upstream checker it was a smaller build option being tested. The internal documentation regarding this step of the release is now updated also.

JacobBarthelmeh commented:
Retest this please Jenkins
